metpy/calc/thermo.py

```diff
@@ -1412,3 +1412,69 @@ def mixed_layer(p, *args, **kwargs):
         ret.append((-1. / actual_depth.m) * np.trapz(datavar_layer, p_layer) *
                    datavar_layer.units)
     return ret
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def dry_static_energy(heights, temperature):
+    r"""Calculate the dry static energy of parcels.
+
+    This function will calculate the dry static energy following the first two terms of
+    equation 3.72 in [REF.
+
+    Notes
+    -----
+    .. math::\text{dry static energy} = c_{pd} * T + gz
+
+    * :math:`T` is temperature
+    * :math:`z` is height
+
+    Parameters
+    ----------
+    heights : array-like
+        Atmospheric height profile
+    temperature : array-like
+        Atmospheric temperature profile
+
+    Returns
+    -------
+    `pint.Quantity`
+        The dry static energy.
+
+    """
+
+    return g * heights + Cp_d * temperature
+
+
+@exporter.export
+@check_units('[length]', '[temperature]', '[dimensionless]')
+def moist_static_energy(heights, temperature, specific_humidity):
+    r"""Calculate the moist static energy of parcels.
+
+    This function will calculate the moist static energy following equation 3.72 in [REF.
```

Comment:
3.72 doesn't exist in my ye olde copy of W&H. You'll have to pull from yours.

Reply:
Do we want two versions of W&H in the docs or should we \<gulp\> update everything to a newer edition?

Reply:
You mean like the two versions we already have in `references.rst`?

Reply:
😁